File: src/main/java/org/goafabric/dbagentnew/dbagent/persistence/PersonRepository.java
package org.goafabric.dbagentnew.dbagent.persistence;

import org.goafabric.dbagentnew.dbagent.persistence.entity.Person;
import org.springframework.data.repository.CrudRepository;

import java.util.List;

public interface PersonRepository extends CrudRepository<Person, String> {

    List<Person> findByFirstNameIgnoreCase(String firstName);

    List<Person> findByLastNameIgnoreCase(String lastName);

    List<Person> findByAddressStreetContainsIgnoreCase(String street);

    List<Person> findByAddressCityContainsIgnoreCase(String city);

    List<Person> findByAllergyAllergyContainsIgnoreCase(String allergy);

}

File: src/main/kotlin/org/goafabric/llm/dbassistant/persistence/PersonRepository.kt
package org.goafabric.llm.dbassistant.persistence

import org.goafabric.llm.dbassistant.persistence.entity.Person
import org.springframework.data.repository.CrudRepository

interface PersonRepository : CrudRepository<Person, String> {
    fun findByFirstNameIgnoreCase(firstName: String): MutableList<Person>

    fun findByLastNameIgnoreCase(lastName: String): MutableList<Person>

    fun findByAddressStreetContainsIgnoreCase(street: String): MutableList<Person>

    fun findByAddressCityContainsIgnoreCase(city: String): MutableList<Person>

    fun findByAllergyAllergyContainsIgnoreCase(allergy: String): MutableList<Person>
}

